Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
-				else
-					m_pFogOfWarCheckbox->SetCheck(1);
+		} else {
+			m_ScenarioScenes = nullptr;
+			m_ActivityLabel->SetText("No Activity selected.");
+			m_DifficultyLabel->SetVisible(false);
+			if (m_DifficultySlider) {
+				m_DifficultySlider->SetVisible(false);
 			}
-			m_pFogOfWarCheckbox->SetEnabled(pSelectedGA->GetFogOfWarSwitchEnabled());
-
-			//Set default clear path to orbit flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultRequireClearPathToOrbit() > -1)
-			{
-				if (pSelectedGA->GetDefaultRequireClearPathToOrbit() == 0)
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(0);
-				else
-					m_pRequireClearPathToOrbitCheckbox->SetCheck(1);
+
+			int textHeight = m_ActivityLabel->ResizeHeightToFit();
+			int padding = 125;
+			m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->Resize(m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox)->GetWidth(), textHeight + padding);
+		}
+
+		KeepBoxInScreenBounds(m_ScenarioCollectionBoxes.at(ScenarioCollections::ActivitySelectBox));
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::UpdatePlayersBox() {
+		RTEAssert(m_SelectedActivity && m_SelectedScene, "Trying to start a scenario game without an activity or a scene.");
+
+		int mouseX = 0;
+		int mouseY = 0;
+		m_GUIInput->GetMousePosition(&mouseX, &mouseY);
+		const GameActivity *gameActivity = dynamic_cast<const GameActivity *>(m_SelectedActivity);
+
+		const GUICollectionBox *hoveredCell = dynamic_cast<GUICollectionBox *>(m_ScenarioGUIController->GetControlUnderPoint(mouseX, mouseY, m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox), 1));
+		if (hoveredCell) {
+			// Find which cell is being hovered over.
+			int hoveredPlayer = PlayerColumns::PlayerColumnCount;
+			int hoveredTeam = TeamRows::TeamRowCount;
+			for (int playerIndex = Players::PlayerOne; playerIndex < PlayerColumns::PlayerColumnCount; ++playerIndex) {
+				for (int teamIndex = Activity::TeamOne; teamIndex < TeamRows::TeamRowCount; ++teamIndex) {
+					if (m_PlayerBoxes.at(playerIndex).at(teamIndex) == hoveredCell) {
+						hoveredPlayer = playerIndex;
+						hoveredTeam = teamIndex;
+					} else if (m_PlayerBoxes.at(playerIndex).at(teamIndex)->GetDrawType() == GUICollectionBox::Color) {
+						// Un-highlight all other cells.
+						m_PlayerBoxes.at(playerIndex).at(teamIndex)->SetDrawColor(c_GUIColorBlue);
+					}
+				}
 			}
-			m_pRequireClearPathToOrbitCheckbox->SetEnabled(pSelectedGA->GetRequireClearPathToOrbitSwitchEnabled());
-
-			//Set default deploy units flag and enable or disable it if necessary
-			if (pSelectedGA->GetDefaultDeployUnits() > -1)
-			{
-				if (pSelectedGA->GetDefaultDeployUnits() == 0)
-					m_pDeployUnitsCheckbox->SetCheck(0);
-				else
-					m_pDeployUnitsCheckbox->SetCheck(1);
+
+			// Make the hovered cell light up and able to be selected if:
```

Comment:
Definitely don't need a paragraph describing what the if check right below it already does :ju:

Also the comment below is junk